crates/polars-time/src/round.rs

```diff
@@ -38,3 +38,22 @@ impl PolarsRound for DateChunked {
             .into_date())
     }
 }
+
+#[cfg(feature = "dtype-duration")]
+impl PolarsRound for DurationChunked {
+    fn round(&self, every: Duration, offset: Duration, _tz: Option<&Tz>) -> PolarsResult<Self> {
+        let (every, offset) = match self.time_unit() {
+            TimeUnit::Nanoseconds => (every.duration_ns(), offset.duration_ns()),
+            TimeUnit::Microseconds => (every.duration_us(), offset.duration_us()),
+            TimeUnit::Milliseconds => (every.duration_ms(), offset.duration_ms()),
+        };
+
+        let out = self.try_apply(|duration| {
```

Comment:
If this function will not fail it is better to use `apply` as that saves a branch.